_posts/2024-07-22-developer-update.md

```diff
@@ -0,0 +1,115 @@
+
+* [[GEOS-11271]](https://osgeo-org.atlassian.net/browse/GEOS-11271) Upgrade spring-security to 5.8
+
+### Spring Security Core implementation of OAuth2 / OIDC
+
+The spring-security-oauth client has reached end-of-life and a GeoServer OAuth2 support must be rewritten or migrated as a result.
+
+There are two paths to migrate to spring-security-core implementation:
+
+* Option: Migrate the existing community module implementations to spring-security-core in place; with as little loss of functionality as possible. This has the advantage of using existing test coverage to maintain a consistent set of functionality.
+
+* Option: Setup a community module alongside the existing implementation with the goal of making a full supported etension. This approach has the advantages of allowing organisations the ability to do A/B testing as both the old and new implementation would be available alongside each other.
+
+Security integrations often require infrastructure to develop and test against, which the core GeoServer team does not have access to for automated tests. While the initial priority will be basic OAuth2 and OIDC parties interested in maintaining support for Google, Azure, GeoNode, etc... are welcome to participate.
+
+Interested Parties:
+
+- Andreas (ITS Digital Solutions GmbH) offered to work on this activity and has indicated they are now ready to start
+- GeoCat is in position to do this work after performing the same upgrade for the GeoNetwork codebase. GeoCat wishes to have an OIDC plugin brought up to full extension status, if financing is available.
+
+Activity: not started
+
+* [[GEOS-11272]](https://osgeo-org.atlassian.net/browse/GEOS-11272) spring-security-oauth replacement, with spring-security 5.8
+
+### ImageN / JAI Replacement
+
+The image processing library used by GeoServer has been donated to the open source community under the name [ImageN](https://projects.eclipse.org/projects/technology.imagen).
+
+We would like to see organisations that depend on GeoServer for earth observation and imagery to step forward with funding for this activity.
+
+Interested Parties: 
+
+* Jody (GeoCat) is available to support this activity, or take lead if funding is available.
+* Andrea (GeoSolutions) has had a deep dive into the implications for the JAI-EXT project outlining a roadmap for project integration
+
+Activities:
+
+* The immediate goal is to add test cases to this codebase and make a version 1.0 release
+
+## Other Development activities
+
+In addition to the key roadmap activities a few other development ideas have been shared:
+
+* OGCAPI Web Services: Gabe (camptocamp) and Jody (GeoCat) are interested in setting up a code sprint for September on this activity.
+  
+  If there is enough interest we would like to arrange a cost-recovery code sprint model as was done for last years API change. Ideally we would like to see a mix of developers, and module lead Andrea (GeoSolutions), attend if sufficient funding is available.
```

Comment:
There is basically zero chance I'll be able to attend any sprint in September, first week I'm in vacation, second week I'm at FOSS4G-NA, the other two weeks I really need to follow up with customer work...

Also, if you intend to go with a rewrite of the modules, a discussion on list would be needed, with an eventual result of having two competing ogcapi sets of modules.

Reply:
I will remove this rather than communicate on gabe's behalf; while I would like to work on OGCAPI I need a plan that is agreed to first, or budget to make a plan.